src/redux/books/books.js

```diff
@@ -0,0 +1,106 @@
+import { call, put, spawn, takeLatest } from "@redux-saga/core/effects";
+import { appID, BASE_URL, deleteData, getData, sendData } from "../../utils";
+
+const ADD_BOOK_REQUEST = 'ADD_BOOK_REQUEST';
+const ADD_BOOK_SUCCESS = 'ADD_BOOK_SUCCESS';
+const REMOVE_BOOK_REQUEST = 'REMOVE_BOOK_REQUEST';
+const REMOVE_BOOK_SUCCESS = 'REMOVE_BOOK_SUCCESS';
+const GET_BOOKS_REQUEST = 'GET_BOOKS_REQUEST';
+const GET_BOOKS_SUCCESS = 'GET_BOOKS_SUCCESS';
+
+
+const initialState = [];
+
+export const addBookRequest = payload => ({
+  type: ADD_BOOK_REQUEST,
+  payload
+})
+export const addBookSuccess = payload => ({
+  type: ADD_BOOK_SUCCESS,
+  payload
+})
+
+export const removeBookRequest = payload => ({
+  type: REMOVE_BOOK_REQUEST,
+  payload
+});
+
+export const removeBookSuccess = payload => ({
+  type: REMOVE_BOOK_SUCCESS,
+  payload
+});
+
+export const getBooksRequest = () => ({
+  type: GET_BOOKS_REQUEST,
+});
+
+export const getBooksSuccess = payload => ({
+  type: GET_BOOKS_SUCCESS,
+  payload
+});
+
+// Redux Saga Generators
+function* addBook({ payload }) {
+  try {
+    yield call(sendData, `${BASE_URL}/apps/${appID}/books`, payload);
+    yield put(addBookSuccess({ ...payload }));
+  } catch (err) {
+    console.log(err);
+  }
+}
```

Comment:
- [ ] Also this function is not a async and await to fetch from the API. please fix

Reply:
This is not even a requirement for this milestone. I had finished before I realized.